sympy/parsing/sympy_parser.py

```diff
@@ -602,7 +602,133 @@ def factorial_notation(tokens, local_dict, global_dict):
             result.append((toknum, tokval))
 
         prevtoken = tokval
+    return result
+
+
+def _transform_equality_operator(tokens, local_dict, global_dict):
+    """Transforms the equality operator ``==`` to instances of Eq.
+
+    This is a helper function for `convert_equality_operators`.
+    Works with expressions containing only one equality operator and no
+    nesting. Expressions like `(1==2)==False` won't work with this
+    and should be used with `convert_equality_operators`.
+
+    Examples: 1==2     to Eq(1,2)
+              1*2==x   to Eq(1*2, x)
+
+    """
+    result = []
+    if (OP, "==") in tokens:
+        result.append((NAME, "Eq"))
+        result.append((OP, "("))
+        for index, token in enumerate(tokens):
+            if token == (OP, "=="):
+                result.append((OP, ","))
+                continue
+            result.append(token)
+        result.append((OP, ")"))
+    else:
+        result = tokens
+    return result
+
+
+def convert_equality_operators(result, local_dict, global_dict):
+    """ Transforms the equality operators ``==`` to instances of Eq.
 
+    Parses the equality operators in the expression and returns
+    appropriate Eq instances.Also works with nested equality operators.
+
+    See also
+    ========
+    convert_equals_signs
+
+    Examples
+    ========
+
+    >>> from sympy.parsing.sympy_parser import (parse_expr,
+    ... standard_transformations, convert_equality_operators)
+    >>> parse_expr("1*2==x", transformations = (
+    ... standard_transformations+
+    ... (convert_equality_operators, )))
+    Eq(2, x)
+    >>> parse_expr("(1*2==x)==False", transformations = (
+    ... standard_transformations+
+    ... (convert_equality_operators, )))
+    Eq(Eq(2, x), False)
+
+    """
+    for step in (_group_parentheses(convert_equality_operators),
+                  _apply_functions,
+                  _transform_equality_operator):
+        result = step(result, local_dict, global_dict)
+
+    result = _flatten(result)
+    return result
+
+
+def _transform_equals_sign(tokens, local_dict, global_dict):
+    """Transforms the equals sign ``=`` to instances of Eq.
```

Comment:
@toolforger #8844 requested this and I asked @asmeurer if it was something that ought to be written for `=` or for `==`. He said that since `parse_expr` is written in a way that makes it easy to plug transformations into it, we could write transformations for both `=` and `==`, and then it's up to the user to use what she wants.

IMO, since `parse_expr` parses any string, we should allow `=` to be parsed, but if the community thinks different, I'll happily remove the functions for `=`.